pincer/objects/guild/guild.py

```diff
@@ -482,6 +484,217 @@ async def modify_member(self, _id: int, **kwargs) -> GuildMember:
+                               access_token: str,
+                               nick: Optional[str] = None,
+                               roles: Optional[List[Snowflake]] = None,
+                               mute: Optional[bool] = None,
+                               deaf: Optional[bool] = None
+                               ) -> Optional[GuildMember]:
+        """|coro|
+        Adds a user to the guild, provided you have a valid oauth2 access token for the user with the guilds.join scope.
+
+        Parameters
+        ----------
+        user_id : str
+            id of the user to be added
+        access_token : str
+            an oauth2 access token granted with the guilds.join to the bot's application for the user you want to add to the guild
+        nick : Optional[str]
+        	value to set users nickname to
+        roles : Optional[List[:class:`~pincer.utils.snowflake.Snowflake`]]
+        	array of role ids the member is assigned
+        mute : Optional[bool]
+        	whether the user is muted in voice channels
+        deaf : Optional[bool]
+        	whether the user is deafened in voice channels
+
+        Returns
+        -------
+        :class:`~pincer.objects.guild.member.GuildMember`
+            If the user is not in the guild
+        None
+            If the user is in the guild
+        """
+
+    async def add_guild_member(self, user_id, **kwargs):
+        data = self._http.put(f"guilds/{self.id}/members/{user_id}",
+                              data=kwargs)
+
+        return GuildMember.from_dict(data) if data else None
+
+    async def modify_current_member(self, nick: str) -> GuildMember:
+        """|coro|
+        Modifies the current member in a guild.
+
+        Parameters
+        ----------
+        nick : str
+            value to set users nickname to
+        """
+        data = self._http.patch(f"guilds/{self.id}/members/@me", {"nick": nick})
+        member = GuildMember.from_dict(data)
+        return member
+
+    async def add_guild_member_role(self, user_id: int, role_id: int) -> None:
+        """|coro|
+        Adds a role to a guild member.
+
+        Parameters
+        ----------
+        user_id : int
+            id of the user to give a role to
+        role_id : int
+            id of a role
+        """
+        data = await self._http.put(
+            f"guilds/{self.id}/{user_id}/roles/{role_id}", {})
+        # TODO: remove the blank dictionary once #233 is fixed
+
+    async def remove_guild_member_role(self, user_id: int,
+                                       role_id: int) -> None:
+        """|coro|
+        Removes a role to a guild member.
+
+        Parameters
+        ----------
+        user_id : int
+            id of the user to remove a role from
+        role_id : int
+            id of a role
+        """
+        await self._http.delete(f"guilds/{self.id}/{user_id}/roles/{role_id}",
+                                {})
```

Comment:
```suggestion
        await self._http.delete(
            f"guilds/{self.id}/{user_id}/roles/{role_id}",
        )
```
This should also support the "X-Audit-Log-Reason" header